keyboards/knops/mini/keymaps/mverteuil/keymap.c

```diff
@@ -0,0 +1,349 @@
+        C(KC_X), C(KC_C), C(KC_V)
+    ),
+
+    /*
+     *   Layer 2 (Spectacles Macros)
+     *	 _LT&__	 _____	 _____
+     *	| Win | | Win | | Win |
+     *	| Next| | Top | |Mximze
+     *	|_1/3_| |_1/2_| |_____|
+     *	 _____	 _____	 _____
+     *	| Win | | Win | | Win |
+     *	| Prev| | Bot | | Next|
+     *	|_1/3_| |_1/2_| |Displ|
+     *
+     */
+    LAYOUT(
+        LT_NXTH,TOPHALF,MAXIMIZ,
+        PRVTHRD,BTMHALF,NXTDSPL
+    ),
+
+    /*
+     *  Layer 3 (Layer Select/Reset)
+     *	 _____	 _____	 _____
+     *	| 	  | | 	  | |     |
+     *	|None | |None | |Reset|
+     *	|_____| |_____| |_____|
+     *	 _____	 _____	 _____
+     *	|Layer| |Layer| |Layer|
+     *  |  0  | |  1  | |  2  |
+     *	|_____| |_____| |_____|
+     *
+     *  Layers 0, 1, and 2 have the top left button held to activate this layer.
+     *  Then press the specific layer to switch to it.
+     *
+     */
+    LAYOUT(
+        _______, _______, RESET,
+        TO(0),   TO(1),   TO(2)
+    )
+
+};
+/* clang-format on */
+
+/*
+ *	Knops Mini LED Numbers:
+ *	 _____	 _____	 _____
+ *	| 	  | | 	  | | 	  |
+ *	|  1  | |  2  | |  3  |    <---
+ *	|_____| |_____| |_____|       |      These LEDs are called 'Switch LEDs'
+ *	 _____	 _____	 _____        |----- To turn on/off these leds, use:
+ *	| 	  | | 	  | | 	  |       |	  set_switch_led( [1-6], [true/false]);
+ *	|  4  | |  5  | |  6  |    <---
+ *	|_____| |_____| |_____|
+ *
+ *	 < 0 >   < 1 >   < 2 >     <---      These front-LEDs are called 'Layer LEDs'
+ *							             To turn one of them on, use:
+ *										 set_layer_led( [0-2] );
+ */
+
+void set_switch_led(int ledId, bool state) {
+    if (state) {
+        switch (ledId) {
+            case 1:
+                writePinHigh(D7);
+                break;
+            case 2:
+                if (readPin(B7)) {
+                    writePinHigh(C6);
+                } else {
+                    writePinHigh(C7);
+                }
+                break;
+            case 3:
+                writePinHigh(D4);
+                break;
+            case 4:
+                writePinHigh(E6);
+                break;
+            case 5:
+                writePinHigh(B4);
+                break;
+            case 6:
+                writePinHigh(D6);
+                break;
+        }
+    } else {
+        switch (ledId) {
+            case 1:
+                writePinLow(D7);
+                break;
+            case 2:
+                if (readPin(B7)) {
+                    writePinLow(C6);
+                } else {
+                    writePinLow(C7);
+                }
+                break;
+            case 3:
+                writePinLow(D4);
+                break;
+            case 4:
+                writePinLow(E6);
+                break;
+            case 5:
+                writePinLow(B4);
+                break;
+            case 6:
+                writePinLow(D6);
+                break;
+        }
+    }
```

Comment:
For this entire thing, you could use `writePin(xx, state)` instead, I think.  https://docs.qmk.fm/#/internals_gpio_control

Reply:
yes, except for the B7/C6/C7 oddness— not sure why it changes based on the state of B7 🤔

Reply:
Well, you could still do so. 

And that block could be: 
```c
writePinLow( readPin(B7) ? C6 : C7, state);
```

I'm not sure that would work, but ... "looks right".

Reply:
Hmm, the device is on my desk at my office— so I will validate tomorrow

Reply:
Generally if I encounter a block like this in python, I will separate the concerns by using a dict mapping switch to pin; is there a similar idiom in C where I can separate the data mapping from the data processing concern?

i.e. can I use something more compact or expressive than a switch/case to get the correct pin from the switch id?

Reply:
Honestly, I'm not sure.

Reply:
Fair enough, just a C noob looking to take any help I can get ;)

Reply:
This works 👍